src/softmax/problem_description.cpp

```diff
@@ -24,59 +24,56 @@
  *
  *******************************************************************************/
 
+#include <miopen/datatype.hpp>
 #include <miopen/softmax/problem_description.hpp>
 #include <miopen/names.hpp>
 
 #include <sstream>
+#include <string_view>
 
 namespace miopen {
 
 namespace softmax {
 
 NetworkConfig ProblemDescription::MakeNetworkConfig() const
 {
-    std::ostringstream ss;
+    std::ostringstream ss(isForward ? "sfmfwd-" : "sfmbwd-");
+
+    // all the tensors must be the same size and types
```

Comment:
@CAHEK7 I suspect we _do have_ the appropriate check (possibly in the ctor of PD), right?

Reply:
@atamazov we do:
https://github.com/ROCm/MIOpen/blob/39ec5d9f67ae68005af8a6f9ffc831190c42b89b/src/include/miopen/softmax/problem_description.hpp#L56

Reply:
Good!